smslibrary/src/test/java/com/eis/smslibrary/SMSManagerTest.java

```diff
@@ -0,0 +1,121 @@
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Class used to test SMSManager using Mocks
+ *
+ * @author Marco Cognolato, Giovanni Velludo
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(SmsManager.class)
+public class SMSManagerTest {
+
+    @Captor
+    private ArgumentCaptor<ArrayList<String>> messageTextCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> peerNumberCaptor;
+
+    private final String VALID_PEER_NUMBER = "+393423541601";
+    private final String VALID_MESSAGE_TEXT = "valid text";
+    private final SMSPeer VALID_PEER = new SMSPeer(VALID_PEER_NUMBER);
+    private final SMSMessage VALID_MESSAGE = new SMSMessage(VALID_PEER, VALID_MESSAGE_TEXT);
+    private final SMSManager managerInstance = SMSManager.getInstance();
+    private static final String EMPTY_TELEPHONE_NUMBER = "";
+    private static final String TOO_SHORT_TELEPHONE_NUMBER = "+39";
+    private static final String TOO_LONG_TELEPHONE_NUMBER = "+39111111111111111111";
+    private static final String NO_COUNTRY_CODE_TELEPHONE_NUMBER = "1111111111";
+
+    @Test
+    public void singletonInstance() {
+        assertEquals(SMSManager.getInstance(), SMSManager.getInstance());
+    }
+
+    @Test()
+    public void validMessage_isSent() {
+        ArrayList<String> messageText = new ArrayList<>(1);
+        messageText.add(VALID_MESSAGE_TEXT);
+
+        SmsManager managerMock = mock(SmsManager.class);
+        when(managerMock.divideMessage(anyString())).thenReturn(messageText);
+
+        PowerMockito.mockStatic(SmsManager.class);
+        when(SmsManager.getDefault()).thenReturn(managerMock);
+
+        managerInstance.sendMessage(VALID_MESSAGE);
+
+        verify(managerMock).sendMultipartTextMessage(peerNumberCaptor.capture(),
+                isNull(String.class), messageTextCaptor.capture(), isNull(ArrayList.class),
+                isNull(ArrayList.class));
+        assertEquals(peerNumberCaptor.getValue(), VALID_PEER_NUMBER);
+        assertEquals(messageTextCaptor.getValue(), messageText);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void nullMessage_throws() {
+        managerInstance.sendMessage(null);
+    }
+
+    @Test(expected = InvalidTelephoneNumberException.class)
+    public void sendMessage_withShortTelephoneNumber_throwsException() {
+        SMSMessage message = new SMSMessage(new SMSPeer(TOO_SHORT_TELEPHONE_NUMBER), VALID_MESSAGE_TEXT);
+        managerInstance.sendMessage(message);
+        Assert.fail("Should have thrown an exception");
```

Comment:
Isn't this obvious? The test will fail if it does not expect the specified exception.

Reply:
Right, I'll remove this